src/main/kotlin/no/nav/familie/klage/kabal/event/BehandlingEventService.kt

```diff
@@ -5,34 +5,63 @@ import no.nav.familie.klage.behandling.StegService
 import no.nav.familie.klage.behandling.domain.Behandling
 import no.nav.familie.klage.behandling.domain.StegType
 import no.nav.familie.klage.fagsak.FagsakRepository
+import no.nav.familie.klage.infrastruktur.config.DatabaseConfiguration.StringListWrapper
 import no.nav.familie.klage.kabal.BehandlingEvent
 import no.nav.familie.klage.kabal.BehandlingEventType
+import no.nav.familie.klage.kabal.KlageresultatRepository
+import no.nav.familie.klage.kabal.domain.Klageresultat
 import no.nav.familie.kontrakter.felles.klage.BehandlingStatus
 import no.nav.familie.prosessering.domene.TaskRepository
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
 import java.util.UUID
 
 @Service
 class BehandlingEventService(
     private val behandlingRepository: BehandlingRepository,
     private val fagsakRepository: FagsakRepository,
     private val taskRepository: TaskRepository,
+    private val klageresultatRepository: KlageresultatRepository,
     private val stegService: StegService
 ) {
 
     private val logger = LoggerFactory.getLogger(javaClass)
 
+    @Transactional
     fun handleEvent(behandlingEvent: BehandlingEvent) {
-        val eksternBehandlingId = UUID.fromString(behandlingEvent.kildeReferanse)
-        val behandling = behandlingRepository.findByEksternBehandlingId(eksternBehandlingId)
 
-        when (behandlingEvent.type) {
-            BehandlingEventType.KLAGEBEHANDLING_AVSLUTTET -> behandleKlageAvsluttet(behandling, behandlingEvent)
-            else -> behandleAnke(behandling, behandlingEvent)
+        val finnesKlageresultat = klageresultatRepository.existsById(behandlingEvent.eventId)
+        if (finnesKlageresultat) {
+            logger.warn("Hendelse fra kabal med eventId: ${behandlingEvent.eventId} er allerede lest - prosesserer ikke hendelse.")
+        } else {
+            logger.info("Prosesserer hendelse fra kabal med eventId: ${behandlingEvent.eventId}")
+            val eksternBehandlingId = UUID.fromString(behandlingEvent.kildeReferanse)
+            val behandling = behandlingRepository.findByEksternBehandlingId(eksternBehandlingId)
+
+            lagreKlageresultat(behandlingEvent, behandling)
+
+            when (behandlingEvent.type) {
+                BehandlingEventType.KLAGEBEHANDLING_AVSLUTTET -> behandleKlageAvsluttet(behandling, behandlingEvent)
+                else -> behandleAnke(behandling, behandlingEvent)
+            }
         }
     }
 
+    private fun lagreKlageresultat(behandlingEvent: BehandlingEvent, behandling: Behandling) {
+        val klageresultat = Klageresultat(
+            eventId = behandlingEvent.eventId,
+            type = behandlingEvent.type,
+            utfall = behandlingEvent.detaljer.utledUtfall(),
+            hendelseTidspunkt = behandlingEvent.detaljer.utledHendelseTidspunkt(),
```

Comment:
Skulle vi skille på type dato? Det er jo forskjell på
klagebehandlingAvsluttet?.avsluttet og ?: ankebehandlingOpprettet?.mottattKlageinstans
, men mulig det er greit å samle disse i en?

...og navngivning: er det riktig å bruke "hendelse"-tidspunk? Kan kanskje forveksles med når vi leste hendelsen/eventet? 

Men har ikke noe bedre forslag i hatten 😝

Reply:
Enig i at `hendelse` kan mistolkes. Tidspunktet er koblet til typen hendelse, så går å koble den til typen, eller splitte opp den i oprettet/avslutet

Reply:
Prøvde å brainstorme litt alene på alternativer til hendelseTidspunkt hvis vi ikke skal dele opp i to, men kom ikke på noe som var særlig bedre.